wgpu-core/src/indirect_validation.rs

```diff
@@ -0,0 +1,369 @@
+use thiserror::Error;
+
+use crate::{
+    device::DeviceError,
+    pipeline::{CreateComputePipelineError, CreateShaderModuleError},
+};
+
+#[derive(Clone, Debug, Error)]
+#[non_exhaustive]
+pub enum CreateDispatchIndirectValidationPipelineError {
+    #[error(transparent)]
+    DeviceError(#[from] DeviceError),
+    #[error(transparent)]
+    ShaderModule(#[from] CreateShaderModuleError),
+    #[error(transparent)]
+    ComputePipeline(#[from] CreateComputePipelineError),
+}
+
+/// This machinery requires the following limits:
+///
+/// - max_bind_groups: 2,
+/// - max_dynamic_storage_buffers_per_pipeline_layout: 1,
+/// - max_storage_buffers_per_shader_stage: 2,
+/// - max_storage_buffer_binding_size: 3 * min_storage_buffer_offset_alignment,
+/// - max_push_constant_size: 4,
+/// - max_compute_invocations_per_workgroup 1
+///
+/// Which are be all indirectly satisfied by `DownlevelFlags::INDIRECT_EXECUTION`.
```

Comment:
typo: `are be` should probably be `are`.

```suggestion
/// Which are all indirectly satisfied by `DownlevelFlags::INDIRECT_EXECUTION`.
```

Reply:
suggestion: This reference is oblique, and I think it might help the reader more if we note that it's supposed to be a _requirement_ for activating this functionality, i.e. (combined with the above typo suggestion, modulo a correct text break/reflow):

```suggestion
/// These are all indirectly satisfied by `DownlevelFlags::INDIRECT_EXECUTION`, which is also required for this module's functionality to work.
```

Reply:
Implemented as agreed with a force-push of c851e983.